data-preprocess.py

```diff
@@ -12,12 +12,12 @@
 
 class DataPreprocessor:
     def __init__(self, args):
-        self.data_path = args.data_path or valohai.inputs('dataset').path()
+        self.data_path = args.data_path
         self.model_max_length = args.model_max_length
         self.tokenizer = args.tokenizer
-        self.train_dataset = load_dataset(self.data_path, split='train')
-        self.eval_dataset = load_dataset(self.data_path, split='validation')
-        self.test_dataset = load_dataset(self.data_path, split='test')
+        self.train_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('train.csv'))
+        self.eval_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('validation.csv'))
+        self.test_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('test.csv'))
```

Comment:
Don't these changes mean that the `data_path` argument isn't used anymore? If so, it should be removed altogether.

However, I think it would be better to allow it, and compose these paths based on it (and default it to the `dataset` input directory)?

Maybe more broadly – what's the reason behind this change? Doesn't this work like the original code did?

Reply:
In the original code the dataset input was pointing to `viggo.py` file, which loaded the data from HF for us. We decided that it's not the intuitive way to load the data, and it is better to load from csv. When loading from csv you should pass the path to the exact file. I don’t see how we can have one `data_path`.
Maybe it will look better if we have three inputs like: train_data, test_data, val_data?
Or I delete `data_path` arg for good?

Reply:
> When loading from csv you should pass the path to the exact file. I don’t see how we can have one `data_path`.

It'd be the path to the directory with the files:

```suggestion
        self.data_path = args.data_path or valohai.inputs('dataset').path()
        self.model_max_length = args.model_max_length
        self.tokenizer = args.tokenizer
        self.train_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'train.csv'))
        self.eval_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'validation.csv'))
        self.test_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'test.csv'))
```